src/feature/FormatDocumentFeature.ts

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+import * as vscode from "vscode";
+import { IFeature } from "../feature";
+import { ILogger } from "../logging";
+import { IConnectionManager } from '../connection';
+import { ConnectionStatus } from '../interfaces';
+import { ISettings } from '../settings';
+import * as messages from '../messages';
+
+class RequestParams implements messages.PuppetFixDiagnosticErrorsRequestParams {
+  documentUri: string;
+  alwaysReturnContent: boolean;
+}
+
+class FormatDocumentProvider {
+  private connectionManager: IConnectionManager = undefined;
+
+  constructor(connectionManager:IConnectionManager) {
+    this.connectionManager = connectionManager;
+  }
+
+  public formatTextEdits(document: vscode.TextDocument, options:vscode.FormattingOptions) : Thenable<vscode.TextEdit[]> {
+    if (this.connectionManager.status !== ConnectionStatus.Running ) {
+      vscode.window.showInformationMessage("Formatting Puppet files is not available as the Language Server is not ready");
```

Comment:
I wonder if this is a good time to change the language of this error?

Reply:
I think for right now, adding 'Please wait and try again. The Puppet extension is still loading...' is good